analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -117,5 +118,77 @@ public void IsPattern_OtherPattern_VisitsBothBranches()
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.True))
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.False));
         }
+
+        [TestMethod]
+        public void RecursivePattern_NoSymbol_DoesNothing()
+        {
+            const string code = @"
+if (arg is { })
+{
+    Tag(""ArgNotNull"", arg);
+}
+Tag(""End"", arg);";
+            var validator = SETestContext.CreateCS(code, ", object arg").Validator;
+            validator.ValidateContainsOperation(OperationKind.RecursivePattern);
+            validator.ValidateTag("ArgNotNull", x => x.Should().BeNull());  // ToDo: Should learn that it is not null too
+            validator.TagValues("End").Should().HaveCount(1).And.OnlyContain(x => x == null);
```

Comment:
```suggestion
            validator.TagValues("End").Should().HaveCount(1).And.AllBe(null);
```
or
```suggestion
            validator.TagValues("End").Should().BeEquivalentTo(new object[] { null });
```

Reply:
I don't have `AllBe` :( I expect this to be changed in the next PR anyway so I'll leave it